### ISSUE TITLE
How to create a model without input_shape?

### ISSUE THREAD

**Issue by rex-yue-wu** [fchollet/keras#3108]
Hi all,

In my text detection application, all I need is a stack of convolution, deconvlution, and pooling layers. In other words, my output shape is the same as the input shape. 

The issue here is my training data are of different dimensions, and this cause a problem when I train this network. Of course, I can resize all my training data into a fixed size and then run training, but this setting may cause a problematic aspect ratio. I also tried to train this network by copying network weights and assigning to a newly created network whenever my training data input shape changes. This approach is very expensive in terms of time, because I need to compile a new model for a new input shape every time. 

I remembered that older version keras does not require input_shape. Is it possible not to specify input_shape in the current version of keras?

Many thanks

**Comment by AdityaGudimella** [comment]
Try passing in None for whatever dimension shapes are unknown. For example, if your input shape is (32, 128) but might vary pass in (None, None)

**Comment by rex-yue-wu** [comment]
I tried different versions of keras. It seems that everything works OK unless the used model contains UpSampling layer.  Below is the error message from my machine.


`
>>> print keras.__version__
1.0.6
>>> if 1:     
...      conv1 = Sequential()
...      conv1.add(ZeroPadding2D((1,1),input_shape=(3,None,None)))
...      conv1.add(Convolution2D(64, 3, 3, activation='relu'))
...      conv1.add(ZeroPadding2D((1,1)))
...      conv1.add(Convolution2D(64, 3, 3, activation='relu'))
...      conv1.add(MaxPooling2D((2,2), strides=(2,2)))     
...      conv1.add(Convolution2D(1,1,1, activation='relu'))
...      conv1.add(UpSampling2D((2,2)))     
... 
_Traceback (most recent call last):
  File "<stdin>", line 9, in <module>
  File "keras/models.py", line 146, in add
    output_tensor = layer(self.outputs[0])
  File "keras/engine/topology.py", line 485, in __call__
    self.add_inbound_node(inbound_layers, node_indices, tensor_indices)
  File "keras/engine/topology.py", line 543, in add_inbound_node
    Node.create_node(self, inbound_layers, node_indices, tensor_indices)
  File "keras/engine/topology.py", line 151, in create_node
    output_shapes = to_list(outbound_layer.get_output_shape_for(input_shapes[0]))
  File "keras/layers/convolutional.py", line 1015, in get_output_shape_for
    self.size[0] * input_shape[2],
TypeError: unsupported operand type(s) for *: 'int' and 'NoneType' _
`

However, everything goes fine if I do not add the last layer. 
`
>>> if 1:     
...      conv1 = Sequential()
...      conv1.add(ZeroPadding2D((1,1),input_shape=(3,None,None)))
...      conv1.add(Convolution2D(64, 3, 3, activation='relu'))
...      conv1.add(ZeroPadding2D((1,1)))
...      conv1.add(Convolution2D(64, 3, 3, activation='relu'))
...      conv1.add(MaxPooling2D((2,2), strides=(2,2)))     
...      conv1.add(Convolution2D(1,1,1, activation='relu'))
...      conv1.compile( 'sgd', 'mse' )
...      conv1.predict( np.random.randn( 1,3, 100, 100 )).shape
...      conv1.predict( np.random.randn( 1,3, 200, 200 )).shape
... 
_(1, 1, 50, 50)
(1, 1, 100, 100)_
`

Does anyone know how to solve my problem? I looked at source code, it seems that MaxPooling and UpSampling are implemented in different ways.
